### DISCUSSION TITLE
How to give users multiple roles with different guards?

### DISCUSSION THREAD

**Discussion by mrmmg** [spatie/laravel-permission#1766] [Q&A]
Hi

I'm developing an admin panel and API (for nuxt js PWA) with laravel. In my project, there are two guards exists:
1. web (default laravel guard)
2. api (uses laravel passport settings)

now I have this roles (with this package):

1. name: **user** | guard_name: **web**
2. name: **admin** | guard_name: **web**
3. name: **adviser** | guard_name: **api**

For example, in my project, a user already has the first role when I want to assign the third role for this user with admin panel that works with laravel web guard but an error occurs:

Spatie\Permission\Exceptions\RoleDoesNotExist
There is no role named `adviser`.

But For the second role, there is no problem.

It seems when I logged in into the admin panel **I can only assign the role with web guard, not api guard**. why exactly?

How can I fix that?

**Comment by erikn69** [comment]
Make your own eloquent model for `model_has_roles` and do the `insert`

**Comment by andrekutianski** [comment]
In my scenario I need that User Model can be associate with roles for multiple guards, to accomplish that I just define `$guard_name` on Model as array with all _guard_names_ that can this can be associated.

```php

namespace App\Models;

Class User extends UserModel
{

// ...

protected array $guard_name = ['api', 'web'];

// ...

}

```

With that, I can assign model to multiple guards.
